File: Client/Trivia/Models/User/UserModel.cs
namespace Trivia.Models.User;

public class UserModel : Raw.User
{
    public bool IsCurrentUser { get; init; }
    
    
    public static UserModel FromUser(Raw.User user, bool isCurrentUser)
    {
        return new UserModel
        {
            Id = user.Id,
            Username = user.Username,
            IsCurrentUser = isCurrentUser
        };
    }
}

File: Server/src/common/infrastructure/model/UserModel.cpp
#include "UserModel.h"

UserModel::UserModel(const long id, const std::string &username) :
    id(id),
    username(username)
{}

bool UserModel::operator==(const UserModel &other) const
{
    return this->id == other.id;
}

bool UserModel::operator<(const UserModel &other) const
{
    return this->id < other.id;
}
